usr/src/uts/common/brand/lx/syscall/lx_socket.c

```diff
@@ -3951,6 +3984,39 @@ lx_getsockopt_tcp(sonode_t *so, int optname, void *optval, socklen_t *optlen)
 		}
 		*optlen = sizeof (int);
 		goto out;
+
+	case LX_TCP_INFO:
+		/*
+		 * We only try to fill in the fields, that we know
+		 * some applications expect.
+		 */
+		if (*optlen < sizeof (lx_tcp_info_t)) {
+			error = EINVAL;
+		} else {
+			*optlen = sizeof (lx_tcp_info_t);
+			bzero(optval, *optlen);
+
+			if (so->so_type != SOCK_STREAM)
+			   goto out; 
+
+			lx_tcp_info_t *ti = (lx_tcp_info_t *)optval;
+			conn_t *con = (struct conn_s *)so->so_proto_handle;
+			tcp_t *tp = con->conn_tcp;
+			ti->tcpi_state = STOL_TCPSTATE(tp->tcp_state);
+			ti->tcpi_rto = tp->tcp_rto_initial;
+			ti->tcpi_last_data_recv = tp->tcp_ts_recent;
+			ti->tcpi_rtt = NSEC2USEC(tp->tcp_rtt_sa);
+			ti->tcpi_rttvar = NSEC2USEC(tp->tcp_rtt_sd);
+			ti->tcpi_snd_ssthresh = tp->tcp_swnd;
```

Comment:
I don't think this is it either. How did you arrive at this?

Reply:
I tried to map the definitions that I saw to the LX related fields, I have added some comments to each variable assignment now.